internals/overlord/cmdstate/handlers.go

```diff
@@ -475,21 +475,16 @@ func (e *execution) controlLoop(execID string, pidCh <-chan int, stop <-chan str
 		}
 
 		if err != nil {
-			logger.Debugf("Exec %s: cannot get next websocket reader for PID %d: %v", execID, pid, err)
-			er, ok := err.(*websocket.CloseError)
-			if !ok {
-				break
-			}
-			if er.Code != websocket.CloseAbnormalClosure {
-				break
-			}
-
-			// If an abnormal closure occurred, kill the attached process.
-			err := unix.Kill(pid, unix.SIGKILL)
-			if err != nil {
-				logger.Noticef("Exec %s: cannot send SIGKILL to pid %d: %v", execID, pid, err)
-			} else {
-				logger.Debugf("Exec %s: sent SIGKILL to pid %d", execID, pid)
+			if websocket.IsUnexpectedCloseError(err, websocket.CloseAbnormalClosure, websocket.CloseGoingAway) {
+				logger.Debugf("Exec %s: cannot get next websocket reader for PID %d: %v", execID, pid, err)
```

Comment:
In my testing, when this error is printed out it's actually a `*websocket.netError`, not a `*websocket.CloseError` ("read unix /var/lib/pebble/default/.pebble.socket->@: use of closed network connection").

So is this actually the logic we want?